File: DKPlugins/DKDuktape/DKScreen.cpp
//https://developer.mozilla.org/en-US/docs/Web/API/Screen

#ifdef USE_DKDuktape 
#include "DK/DKApp.h"
#include "DKDuktape/DKScreen.h"


bool DKScreen::Init(){
	DKDEBUGFUNC();

	//Properties
	DKDuktape::AttachFunction("CPP_DKScreen_availTop", DKScreen::availTop);
	DKDuktape::AttachFunction("CPP_DKScreen_availLeft", DKScreen::availLeft);
	DKDuktape::AttachFunction("CPP_DKScreen_availHeight", DKScreen::availHeight);
	DKDuktape::AttachFunction("CPP_DKScreen_availWidth", DKScreen::availWidth);
	DKDuktape::AttachFunction("CPP_DKScreen_colorDepth", DKScreen::colorDepth);
	DKDuktape::AttachFunction("CPP_DKScreen_height", DKScreen::height);
	DKDuktape::AttachFunction("CPP_DKScreen_left", DKScreen::left);
	DKDuktape::AttachFunction("CPP_DKScreen_orientation", DKScreen::orientation);
	DKDuktape::AttachFunction("CPP_DKScreen_pixelDepth", DKScreen::pixelDepth);
	DKDuktape::AttachFunction("CPP_DKScreen_top", DKScreen::top);
	DKDuktape::AttachFunction("CPP_DKScreen_width", DKScreen::width);
	DKDuktape::AttachFunction("CPP_DKScreen_mozEnabled", DKScreen::mozEnabled);
	DKDuktape::AttachFunction("CPP_DKScreen_mozBrightness", DKScreen::mozBrightness);

	//Methods
	DKDuktape::AttachFunction("CPP_DKScreen_lockOrientation", DKScreen::lockOrientation);
	DKDuktape::AttachFunction("CPP_DKScreen_unlockOrientations", DKScreen::unlockOrientation);

	DKClass::DKCreate("DKDuktape/DKScreen.js");
	return true;
}


//Properties
int DKScreen::availTop(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::availLeft(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::availHeight(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::availWidth(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::colorDepth(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::height(duk_context* ctx){
	int height;
	if(!DKUtil::GetScreenHeight(height)){ return 0; }
	duk_push_number(ctx, height);
	return 1;
}

int DKScreen::left(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::orientation(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::pixelDepth(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::top(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::width(duk_context* ctx){
	int width;
	if(!DKUtil::GetScreenWidth(width)) { return 0; }
	duk_push_number(ctx, width);
	return 1;
}

int DKScreen::mozEnabled(duk_context* ctx){
	//TODO
	return false;
}

int DKScreen::mozBrightness(duk_context* ctx){
	//TODO
	return false;
}



//Methods
int DKScreen::lockOrientation(duk_context* ctx) {
	//TODO - https://developer.mozilla.org/en-US/docs/Web/API/Screen/lockOrientation
	DKString orientation = duk_require_string(ctx, 0);
	return false;
}

int DKScreen::unlockOrientation(duk_context* ctx){
	//TODO - https://developer.mozilla.org/en-US/docs/Web/API/Screen/unlockOrientation
	return false;
}

#endif //USE_DKDuktape

File: DKPlugins/DKDuktape/DKScreen.js
//https://developer.mozilla.org/en-US/docs/Web/API/Screen

var Screen = function(pointer) {
    Object.defineProperty(this, "availTop", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_availTop(this.pointer);
        }
    });
    Object.defineProperty(this, "availLeft", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_availLeft(this.pointer);
        }
    });
    Object.defineProperty(this, "availHeight", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_availHeight(this.pointer);
        }
    });
    Object.defineProperty(this, "availWidth", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_availWidth(this.pointer);
        }
    });
    Object.defineProperty(this, "colorDepth", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_colorDepth(this.pointer);
        }
    });
    Object.defineProperty(this, "height", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_height(this.pointer);
        }
    });
    Object.defineProperty(this, "left", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_left(this.pointer);
        }
    });
    Object.defineProperty(this, "orientation", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_orientation(this.pointer);
        }
    });
    Object.defineProperty(this, "pixelDepth", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_pixelDepth(this.pointer);
        }
    });
    Object.defineProperty(this, "top", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_top(this.pointer);
        }
    });
    Object.defineProperty(this, "width", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_width(this.pointer);
        }
    });
    Object.defineProperty(this, "mozEnabled", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_mozEnabled(this.pointer);
        }
    });
    Object.defineProperty(this, "mozBrightness", {
        configurable: true,
        get: function() {
            return CPP_DKScreen_mozBrightness(this.pointer);
        }
    });

    Screen.prototype.lockOrientation = function(orientation) {
        return CPP_DKScreen_lockOrientation(orientation);
    }
    Screen.prototype.unlockOrientation = function() {
        return CPP_DKScreen_unlockOrientation();
    }

    return EventTarget.call(this, pointer);
}
Screen.prototype = EventTarget.prototype;

var screen = new Screen("screen");
